symengine/cwrapper.cpp

```diff
@@ -58,6 +58,25 @@ inline bool is_aligned(T *p, size_t n = alignof(T))
 
 extern "C" {
 
+
+#define CWRAPPER_BEGIN()                                                       \
+    try                                                                        \
+    {
+
+#define CWRAPPER_END(error, code)                                              \
+        return 0;                                                              \
+    }                                                                          \
+    catch ( error )                                                            \
+    {                                                                          \
+        return code;                                                           \
+    }
+
+#define CWRAPPER_END_EXTRA(error, code)                                        \
```

Comment:
@certik I'm going to make SymEngine specific exceptions, and add them to relevant wrappers. (Eg: `DivisionByZero` for `div`, different types of `ParseError` for `parse` etc.). I'm yet to add them. I will include this also in the next commit.

@isuruf suggested that doing this and returning different error codes, so that in the Ruby Wrapper we can show error messages which makes more sense to the user.